### PR TITLE
[6561] vcsparser publish code churn to cosmosdb

### PR DESCRIPTION
**Pull request by sebastiansztangierski**
1. Changed the required target framework to 4.8.
2. Added a new way of publishing CodeChurn and BugDatabase data to the Azure Cosmos database.
3. Added a new way of downloading data based on a date range.  This will output files to JSON just like a standard output processor does.
4. Added a new way of getting CodeChurn/BugDatabase data per file.
5. Added logic for querying to cosmos db, this includes getting documents by date range, creating new documents and deleting multiple documents.
6. Added unit tests.

### CODE REVIEW

**Review by roryclaasen — APPROVED** [review]
Looks good to me

File: src/vcsparser.core/vcsparser.core.csproj [hunk 1]
```diff
@@ -29,28 +32,173 @@
...
     <Reference Include="System.ComponentModel.Composition" />
+    <Reference Include="System.Configuration" />
+    <Reference Include="System.Configuration.ConfigurationManager, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Configuration.ConfigurationManager.4.7.0\lib\net461\System.Configuration.ConfigurationManager.dll</HintPath>
+    </Reference>
     <Reference Include="System.Console, Version=4.0.1.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <HintPath>..\packages\System.Console.4.3.1\lib\net46\System.Console.dll</HintPath>
     </Reference>
     <Reference Include="System.Core" />
+    <Reference Include="System.Data.OracleClient" />
```
Comment (ericlemes) [inline src/vcsparser.core/vcsparser.core.csproj:88]: OracleClient?

File: src/vcsparser.core/vcsparser.core.csproj [hunk 2]
```diff
@@ -29,28 +32,173 @@
...
+    <Reference Include="Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Bcl.AsyncInterfaces.1.0.0\lib\net461\Microsoft.Bcl.AsyncInterfaces.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Bcl.HashCode, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Bcl.HashCode.1.1.0\lib\net461\Microsoft.Bcl.HashCode.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Win32.Primitives, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Win32.Primitives.4.0.1\lib\net46\Microsoft.Win32.Primitives.dll</HintPath>
+    </Reference>
+    <Reference Include="MongoDB.Bson, Version=2.4.4.0, Culture=neutral, PublicKeyToken=15b1115599983c50, processorArchitecture=MSIL">
```
Comment (ericlemes) [inline src/vcsparser.core/vcsparser.core.csproj:57]: MongoDB?

File: src/vcsparser.core/vcsparser.core.csproj [hunk 3]
```diff
@@ -29,28 +32,173 @@
...
     </Reference>
     <Reference Include="System.Core" />
+    <Reference Include="System.Data.OracleClient" />
+    <Reference Include="System.Diagnostics.DiagnosticSource, Version=4.0.4.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Diagnostics.DiagnosticSource.4.6.0\lib\net46\System.Diagnostics.DiagnosticSource.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Diagnostics.TraceSource, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Diagnostics.TraceSource.4.0.0\lib\net46\System.Diagnostics.TraceSource.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Drawing" />
```
Comment (ericlemes) [inline src/vcsparser.core/vcsparser.core.csproj:95]: System.Drawing?

**Review by ericlemes — CHANGES_REQUESTED** [review]
Overall, it looks very good. There's just a few suspect references that seems unnecessary at first look. Can you please double checked if they are really needed?

File: src/vcsparser.core/vcsparser.core.csproj (hunk 2 above)
Comment (roryclaasen) [reply src/vcsparser.core/vcsparser.core.csproj:57]: If I remember correctly, this gets added by add the cosmos package.
Worth a double check though

File: src/vcsparser.core/vcsparser.core.csproj (hunk 1 above)
Comment (sebastiansztangierski) [reply src/vcsparser.core/vcsparser.core.csproj]: Removed.

**Review by roryclaasen — CHANGES_REQUESTED** [review]
@sebastiansztangierski the [CI tasks](https://github.com/ericlemes/vcsparser/blob/master/appveyor.yml) now fail due to the updated packages. Can you point OpenCover to the "4.7.1221" folder to fix it

**Review by roryclaasen — APPROVED** [review]

**Review by ericlemes — APPROVED** [review]

**Pull request merged**